doc/source/data/accessing-datasets.rst

```diff
@@ -4,47 +4,188 @@
+that we may want to split into separate tensors. By informing ``ds.to_tf()`` of the
+label column, it will yield ``(features, label)`` tensor pairs for each batch.
+
+.. literalinclude:: ./doc_code/accessing_datasets.py
+  :language: python
+  :start-after: __tf_with_label_begin__
+  :end-before: __tf_with_label_end__
+
+The types of the label and feature columns will be inferred from the data by default;
+these can be overridden with the ``label_column_dtype`` and ``feature_column_dtypes``
+args.
 
-    @ray.remote
-    def consume(data: Dataset[int]) -> int:
-        num_batches = 0
-        for batch in data.iter_batches():
-            num_batches += 1
-        return num_batches
+By default, all feature columns will be concatenated into a single tensor; however,
+depending on the structure of the ``feature_columns`` argument, you can also get feature
+column batches as a list of tensors or a dict of tensors (with one or more column in
+each tensor). See the :meth:`.to_tf() API docs <ray.data.Dataset.to_tf>` for
+details.
 
-    ds = ray.data.range(10000)
-    ray.get(consume.remote(ds))
-    # -> 200
+.. note::
 
-Datasets can be split up into disjoint sub-datasets.
-Locality-aware splitting is supported if you pass in a list of actor handles to the ``split()`` function along with the number of desired splits.
-This is a common pattern useful for loading and splitting data between distributed training actors:
+  If we have tensor feature columns (where each item in the column is an multi-dimensional
+  tensor) and any of the feature columns are different shapes, these columns are
+  incompatible and we will not be able to stack the column tensors into a single tensor.
+  Instead, we will need to group the columns by compatibility in the ``feature_columns``
+  argument.
 
-.. code-block:: python
+  Check out the :ref:`tensor data feature guide <datasets_tensor_ml_exchange>` for more
+  information on how to handle this.
 
-    @ray.remote(num_gpus=1)
-    class Worker:
-        def __init__(self, rank: int):
-            pass
+Splitting Into and Consuming Shards
+===================================
 
-        def train(self, shard: ray.data.Dataset[int]) -> int:
-            for batch in shard.iter_batches(batch_size=256):
-                pass
-            return shard.count()
+Datasets can be split up into disjoint sub-datasets, or shards.
+Locality-aware splitting is supported if you pass in a list of actor handles to the
+:meth:`ds.split() <ray.data.Dataset.split>` function along with the number of desired splits.
+This is a common pattern useful for loading and sharding data between distributed training actors:
 
-    workers = [Worker.remote(i) for i in range(16)]
-    # -> [Actor(Worker, ...), Actor(Worker, ...), ...]
+.. literalinclude:: ./doc_code/accessing_datasets.py
+  :language: python
+  :start-after: __split_begin__
+  :end-before: __split_end__
```

Comment:
Should we add a note that if using Ray Datasets with Ray Train, the splitting will automatically done for you? There was a user who was confused on this.